ambry-store/src/test/java/com.github.ambry.store/BlobStoreTest.java

```diff
@@ -667,11 +670,24 @@ public void shutdownTest() throws StoreException {
     verifyOperationFailuresOnInactiveStore(store);
     StorageManagerMetrics metrics = new StorageManagerMetrics(new MetricRegistry());
     StoreConfig config = new StoreConfig(new VerifiableProperties(properties));
-    store = new BlobStore(storeId, config, scheduler, storeStatsScheduler, diskIOScheduler, metrics, tempDirStr,
-        LOG_CAPACITY, STORE_KEY_FACTORY, recovery, hardDelete, time);
+    store = new BlobStore(storeId, config, scheduler, storeStatsScheduler, diskIOScheduler, diskSpaceAllocator, metrics,
+        tempDirStr, LOG_CAPACITY, STORE_KEY_FACTORY, recovery, hardDelete, time);
     verifyOperationFailuresOnInactiveStore(store);
   }
 
+  @Test
+  public void diskSpaceRequirementsTest() throws StoreException {
+    DiskSpaceRequirements requirements = store.getDiskSpaceRequirements();
+    if (!isLogSegmented) {
+      assertNull("Expected null DiskSpaceRequirements for non segmented log", requirements);
+      return;
+    }
+    assertEquals(SEGMENT_CAPACITY, requirements.getSegmentSizeInBytes());
+    // expect three log segments to be already allocated (from setup process)
+    assertEquals((LOG_CAPACITY / SEGMENT_CAPACITY) - 3, requirements.getSegmentsNeeded());
+    assertEquals(0, requirements.getSwapSegmentsInUse());
```

Comment:
added temp file test. @nsivabalan There are tests for changes in allocated segments inside of LogTest, but I added a couple test cases where segments are added.